runtime/near-vm-logic/src/logic.rs

```diff
@@ -2481,7 +2554,19 @@ impl<'a> VMLogic<'a> {
     }
 
     /// Computes the outcome of execution.
-    pub fn outcome(self) -> VMOutcome {
+    #[allow(unused_mut)]
+    pub fn outcome(mut self) -> VMOutcome {
+        #[cfg(feature = "protocol_feature_function_call_weight")]
+        if !self.context.is_view() {
+            // Distribute unused gas to scheduled function calls
+            let unused_gas = self.context.prepaid_gas - self.gas_counter.used_gas();
+
+            // Distribute the unused gas and prepay for the gas.
+            if self.ext.distribute_unused_gas(unused_gas) {
+                self.gas_counter.prepay_gas(unused_gas).unwrap();
+            }
+        }
```

Comment:
This function now has a nontrivial side effect. We might want to rename it and add a comment about what it does

Reply:
took a stab at this, open to any suggestions of naming or any other information to include in the rustdocs since it's opinionated